File: src/Dto/Project.Dto.Http/PostHistory/PostHistoryDto.cs
namespace Project.Dto.Http.PostHistory;

public class PostHistoryDto
{
    public PostHistoryDto(
        Guid postId,
        Guid employeeId,
        DateOnly startDate,
        DateOnly? endDate)
    {
        PostId = postId;
        EmployeeId = employeeId;
        StartDate = startDate;
        EndDate = endDate;
    }

    public Guid PostId { get; set; }
    public Guid EmployeeId { get; set; }
    public DateOnly StartDate { get; set; }
    public DateOnly? EndDate { get; set; }
}

File: front/src/app/dto/post-history.dto.ts
/**
 * DTO for post history received from API
 */
export interface PostHistoryDto {
  postId: string;
  employeeId: string;
  startDate: string; // ISO date string (DateOnly from backend)
  endDate?: string; // ISO date string (DateOnly from backend)
}

/**
 * DTO for creating a post history
 */
export interface CreatePostHistoryDto {
  postId: string;
  employeeId: string;
  startDate: string; // ISO date string (DateOnly from backend)
  endDate?: string; // ISO date string (DateOnly from backend)
}

/**
 * DTO for updating a post history
 */
export interface UpdatePostHistoryDto {
  startDate?: string; // ISO date string (DateOnly from backend)
  endDate?: string; // ISO date string (DateOnly from backend)
}

